inject/src/main/java/io/micronaut/context/DefaultApplicationContext.java

```diff
@@ -712,18 +713,29 @@ public Environment stop() {
 
         @Override
         public Environment start() {
-            if (isRuntimeConfigured && bootstrapEnvironment == null) {
-                bootstrapEnvironment = createBootstrapEnvironment(getActiveNames().toArray(new String[0]));
+            String bootstrapContextProp = System.getProperty(Environment.BOOTSTRAP_CONTEXT_PROPERTY);
+            Boolean bootstrapEnabled = bootstrapContextProp == null ? null : Boolean.parseBoolean(bootstrapContextProp);
+            if (bootstrapEnabled == null) {
+                bootstrapEnabled = this.bootstrapEnabled;
+            }
+            if (bootstrapEnvironment == null && (bootstrapEnabled == null || bootstrapEnabled)) {
+                BootstrapEnvironment bootstrapEnv = createBootstrapEnvironment(getActiveNames().toArray(new String[0]));
+                if (bootstrapEnabled != null || CollectionUtils.isNotEmpty(bootstrapEnv.readPropertySourceList(bootstrapEnv.getPropertySourceRootName()))) {
+                    bootstrapEnvironment = startBootstrapEnvironment(bootstrapEnv);
+                } else {
+                    if (LOG.isInfoEnabled()) {
+                        LOG.info("The bootstrap environment is enabled but no local property sources were found. Consider disabling the bootstrap environment to gain startup time.");
```

Comment:
The message needs a link to the reference documentation that explains how this can be disabled. From a user perspective most users probably don't even know what the boostrap environment is let alone how to disable it.